lib/Notes/NoteEditPage/NoteEditPage.js

```diff
@@ -0,0 +1,192 @@
+
+import {
+  stripesConnect,
+  TitleManager,
+} from '@folio/stripes-core';
+
+import { Icon } from '@folio/stripes-components';
+
+import NoteForm from '../NoteForm';
+import { noteTypesCollectionShape } from '../response-shapes';
+
+@stripesConnect
+class NoteEditPage extends Component {
+  static propTypes = {
+    domain: PropTypes.string.isRequired,
+    entityTypePluralizedTranslationKeys: PropTypes.objectOf(PropTypes.string),
+    entityTypeTranslationKeys: PropTypes.objectOf(PropTypes.string),
+    mutator: PropTypes.shape({
+      note: PropTypes.shape({
+        PUT: PropTypes.func.isRequired,
+      }).isRequired,
+    }).isRequired,
+    navigateBack: PropTypes.func.isRequired,
+    noteId: PropTypes.string.isRequired,
+    paneHeaderAppIcon: PropTypes.string.isRequired,
+    referredEntityData: PropTypes.shape({
+      id: PropTypes.string.isRequired,
+      name: PropTypes.string.isRequired,
+      type: PropTypes.string.isRequired,
+    }),
+    resources: PropTypes.shape({
+      note: PropTypes.object,
+      noteTypesData: PropTypes.shape({
+        records: PropTypes.arrayOf(noteTypesCollectionShape).isRequired,
+      }),
+    }).isRequired,
+  }
+
+  static manifest = Object.freeze({
+    noteTypesData: {
+      type: 'okapi',
+      path: 'note-types',
+    },
+    note: {
+      type: 'okapi',
+      path: 'notes/!{noteId}',
+    },
+  });
+
+  state = {
+    submitIsPending: false,
+    submitSucceeded: false,
+  };
+
+  onSubmit = async (noteData) => {
+    try {
+      await this.sendNoteData(noteData);
+      this.handleSuccessResponse();
+    } catch (err) {
+      this.handleFailedResponse();
+    }
+  }
+
+  sendNoteData(noteData) {
+    this.setState({ submitIsPending: true });
+
+    return this.props.mutator.note.PUT(this.serializeNoteData(noteData));
+  }
+
+  serializeNoteData = (formData) => {
+    const {
+      domain,
+      resources,
+    } = this.props;
+
+    const { links } = resources.note.records[0];
+
+    return {
+      domain,
+      typeId: formData.type,
+      content: formData.content,
+      title: formData.title,
+      links,
+    };
+  }
+
+  handleSuccessResponse() {
+    this.setState({
+      submitSucceeded: true,
+      submitIsPending: false,
+    }, this.props.navigateBack);
+  }
+
+  handleFailedResponse() {
+    this.setState({ submitIsPending: false });
+  }
+
+  renderSpinner() {
+    return (
+      <Icon icon="spinner-ellipsis" />
+    );
+  }
+
+  getNoteTypesSelectOptions = () => {
+    const { noteTypes } = this.props.resources.noteTypesData.records[0];
+
+    return noteTypes.map(noteType => ({
+      label: noteType.name,
+      value: noteType.id
+    }));
+  }
+
+  getNoteFormData() {
+    const {
+      typeId: type,
+      content,
+      title,
+      links,
+    } = this.props.resources.note.records[0];
+
+    return {
+      type,
+      content,
+      title,
+      links,
+    };
+  }
+
+  getNoteMetadata() {
+    const {
+      createdDate,
```

Comment:
Metadata props are different from the metadata component expects for.
So you will need to map the props.
Like this:
const createdBy = get(note, 'records[0].metadata.createdByUsername');
    const createdDate = get(note, 'records[0].metadata.createdDate');
    const lastUpdatedBy = get(note, 'records[0].metadata.updatedByUsername');
    const lastUpdatedDate = get(note, 'records[0].metadata.updatedDate');
    const metadata = {
      contentId: 'noteEditMetadataContent',
      createdBy,
      createdDate,
      id: 'noteEditMetadata',
      lastUpdatedBy,
      lastUpdatedDate,
      showUserLink: false,
    };